automation/src/main/java/org/greenplum/pxf/automation/structures/tables/pxf/ForeignTable.java

```diff
@@ -47,8 +47,8 @@ protected String createOptions() {
         // foreign tables do not have locations, parameters go into options
         // path (resource option for FDW) should always be present
         StringJoiner joiner = new StringJoiner(",", " OPTIONS (", ")");
-        // TODO Add Escape E to the URI like external table It will help cases like HcfsGlobbingTest
-        appendOption(joiner,"resource", getPath(), true);
+        // Adding Escape E here in the resource just like external table in the URI.
+        appendOption(joiner,"resource ", "E'"+getPath()+"'", false);
```

Comment:
instead of doing this unconditionally, can we use the existing pattern and do here:
```suggestion
        appendOption(joiner,"resource ", getPath(), !getPath().startsWith("E"));
```
and just use `E'whatever'` in the actual test where we need to escape the path ?